src/Config/Converters/DatasourceHealthOptionsConvertorFactory.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Config.ObjectModel;
+
+namespace Azure.DataApiBuilder.Config.Converters;
+
+internal class DatasourceHealthOptionsConvertorFactory : JsonConverterFactory
+{
+    /// <inheritdoc/>
+    public override bool CanConvert(Type typeToConvert)
+    {
+        return typeToConvert.IsAssignableTo(typeof(DatasourceHealthCheckConfig));
+    }
+
+    /// <inheritdoc/>
+    public override JsonConverter? CreateConverter(Type typeToConvert, JsonSerializerOptions options)
+    {
+        return new HealthCheckOptionsConverter();
+    }
+
+    private class HealthCheckOptionsConverter : JsonConverter<DatasourceHealthCheckConfig>
+    {
+        /// <summary>
+        /// Defines how DAB reads an entity's health options and defines which values are
+        /// used to instantiate EntityHealthCheckConfig.
+        /// </summary>
+        /// <exception cref="JsonException">Thrown when improperly formatted health check options are provided.</exception>
+        public override DatasourceHealthCheckConfig? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.Null)
+            {
+                return new DatasourceHealthCheckConfig() { Enabled = true, Name = null, ThresholdMs = 1000 };
```

Comment:
Seems like literal constants like this should be moved out to a constant so it can be reused and named.